templates/aws-tap-entrypoint-existing-vpc.template.yaml

```diff
@@ -2328,230 +2328,152 @@ Resources:
         Fn::Base64: !Sub
           - |
             #!/bin/bash
-            set -xe
-            user=ubuntu
-            arch=$(dpkg --print-architecture)
-            uname=$(uname -m)
-            tap_dir=/home/$user/tap-setup-scripts
+            set -e
+            set -u
+            set -o pipefail
+
+            set -x
+
             cat <<EOF >> /etc/environment
             AWS_REGION=${AWS::Region}
```

Comment:
Add all AWS input parameters in this file /etc/environment

Reply:
Are you saying you want to push [all of these](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/pull/48/files#diff-0d891330f3a704011d4ebc74cd8faaf8a91d3431e2a5fdcfbcf9a8be544b8515R2350-R2389) into `/etc/environment`?

I don't think we want to expose all those for all session?

Reply:
> [...] but our org has a clear stance that it is not secure. At the very least don't put anything sensitive in env vars.

I guess this is mainly a comment on generally setting some vars for (a subset of) users (e.g. via `/etc/environment`) and not so much about handing off some data/config/... to subprocesses. Is that right?

FWIW:
- I'd push for not using `/etc/environment` at all, except maybe for those vars which are already in there (`AWS_DEFAULT_REGION` & `AWS_REGION`) and I think are probably uncontroversial(?)
- I haven't seen any sensitive var in [the list we expose via exports][vars] to the sub-process(es)
- Still, for those I have explicitly disabled `xtrace` (even though they might be logged later in the spawned script, which I pretty much just copied from the original inline'd user-data)
- If we don't want to export [these vars][vars] we can also `source` the externalised cloud-init instead of forking it.
  However, if we are that concerned we might also need to look into how those vars are used later on (e.g. are they used as commandline args and potentially visible via `ps auxf` or other means?).
  But generally I wouldn't be mad about sourcing, don't see much issues with that.
- There is TAPPC-163 which talks about potentially different ways how to handle all these configs/settings/data/...

[vars]: https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/pull/48/files#diff-0d891330f3a704011d4ebc74cd8faaf8a91d3431e2a5fdcfbcf9a8be544b8515R2344-R2390

Reply:
cool - I'm fine with taking a deeper look in TAPPC-163 and not making any decisions as part of this PR.